### ISSUE TITLE
copy MutableMatrix

### ISSUE THREAD

**Issue by antonleykin** [Macaulay2/M2#364]
Let A be a MutableMatrix. Then the following lines seem to have the same effect:
```
B = A
B = copy A
```
... while the expected behavior of the second line is "copy x yields a copy of x".

**Comment by DanGrayson** [comment]
Yes, "copy" should copy it or give an error.  The documentation states that "copy" works
only for hash tables, arrays, lists or sequences, so the documentation might have to be changed,
too.

**Comment by DanGrayson** [comment]
A work-around is available as `B = mutableMatrix matrix A`.

**Comment by DanGrayson** [comment]
It turns out that a mutable matrix *is* a hashtable, and "copy" works as advertised on it.  But the counterintuitive aspect is that one of the entries in the hash table is a pointer to the "raw" mutable matrix, and it's the entries of that that get modified.

So, it's a bit counterintuitive, and it's not easy to make it work as desired, since "copy" is a low level operation.  I'm not sure whether there is something sensible to do about this.

One idea would be to make "copy" do something sensible to raw mutable matrices:
```
--- a/M2/Macaulay2/d/hashtables.dd
+++ b/M2/Macaulay2/d/hashtables.dd
@@ -873,6 +873,7 @@ export copy(e:Expr):Expr := (
      when e
      is a:List do if a.Mutable then Expr(copy(a)) else e
      is o:HashTable do if o.Mutable then Expr(copy(o)) else e
+     is RawMutableMatrixCell do buildErrorPacket("copy not yet implemented for mutable matrices")
      else e);
 setupfun("copy",copy);
 
```

and then incorporate that functionality into something new of higher level.

Further ideas are welcome.

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

Still reproduces on 1.26.06, and #944 turns out to be the same defect through a different
operator, so it is a duplicate of this issue and the two are being tracked here together.

`new MutableMatrix from M` aliases exactly as `copy M` does:

```
M = mutableMatrix{{1,2},{3,4}}
N = new MutableMatrix from M
N_(0,0) = 55

  M_(0,0)   is now 55
```

@DanGrayson's diagnosis above covers both — the hash table holds a pointer to the
`RawMutableMatrix` and it is the contents of that which get modified — but there is one thing
specific to `new … from` worth recording here: there is no `NewFromMethod` for
`(MutableMatrix, MutableMatrix)` at all. `lookup(NewFromMethod, MutableMatrix, MutableMatrix)` is
`null`, whereas `new MutableMatrix from Matrix` does exist, at `mutablemat.m2:34`, and copies
properly by going through `rawMutableMatrix raw m`.

So whatever is decided here should cover both entry points: `copy` needs to duplicate or refuse,
and `new … from` needs a method that does the same. The workaround in this thread,
`mutableMatrix matrix A`, still deep-copies and still works for either.

<sub>Triaged in a sweep of the open issues; source of truth: [`bug-triage/issues.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/issues.tsv).</sub>
